src/foam/u2/view/OverlayActionListView.js

```diff
@@ -0,0 +1,172 @@
+    },
+    {
+      type: 'URL',
+      name: 'imageURL_',
+      expression: function(restingImageURL, hoverImageURL, disabledImageURL, activeImageURL, hovered_, disabled_, active_) {
+        if ( disabled_ ) {
+          return `url(${disabledImageURL})`;
+        } else if ( active_ ) {
+          return `url(${activeImageURL})`;
+        } else if ( hovered_ ) {
+          return `url(${hoverImageURL})`;
+        }
+        return `url(${restingImageURL})`;
+      }
+    },
+    {
+      type: 'Boolean',
+      name: 'hovered_'
+    },
+    {
+      type: 'Boolean',
+      name: 'disabled_'
+    },
+    {
+      type: 'Boolean',
+      name: 'active_'
+    }
+  ],
+
+  css: `
+    ^action {
+      padding: 10px;
+    }
+
+    ^action.disabled {
+      color: #aaa;
+    }
+
+    ^action:hover:not(.disabled) {
+      cursor: pointer;
+      background-color: %ACCENTCOLOR%;
+    }
+
+    ^icon {
+      background-size: 24px;
+      width: 24px;
+      height: 24px;
+    }
+
+    ^icon:hover {
+      cursor: pointer;
+    }
+  `,
+
+  methods: [
+    function initE() {
+      var self = this;
+
+      var overlay = this.OverlayDropdown.create();
+      this.active_$.follow(overlay.opened$);
+      this.disabled_$.follow(foam.core.ExpressionSlot.create({
+        args: this.data.map((a) => a.createIsAvailable$(this.obj$)),
+        code: (...rest) => ! rest.reduce((l, r) => l || r, false)
+      }));
+
+      this.
+        callIf(this.data.length > 0, function() {
+          overlay.forEach(this.data, function(action) {
+            this.
+              start().
+                show(action.createIsAvailable$(self.ConstantSlot.create({
+                  value: self.obj
+                }))).
+                addClass(self.myClass('action')).
+                add(action.label).
+                call(async function() {
+                  if ( await action.isEnabledFor(self.obj) ) {
+                    this.on('click', function(evt) {
+                      overlay.close();
+                      action.maybeCall(self.obj.__subContext__, self.obj);
+                    });
+                  } else {
+                    this.addClass('disabled');
+                  }
+                }).
+              end();
+          });
+
+          // Add the overlay to the controller so if the table is inside a
+          // container with `overflow: hidden` then this overlay won't be cut
+          // off.
+          this.ctrl.add(overlay);
+        }).
+
+        start('span').
+          addClass('noselect').
+          start().
+            addClass(this.myClass('icon')).
+            style({ 'background-image': this.imageURL_$ }).
+            on('mouseover', this.onMouseOver).
+            on('mouseout', this.onMouseOut).
+            callIf(this.data.length > 0, function() {
+              this.on('click', function(evt) {
+                if ( self.disabled_ ) return;
+                overlay.open(evt.clientX, evt.clientY);
```

Comment:
I would move building the overlay contents down here.  No sense building all the action views for the overlay until we're about to render it.  Otherwise we're spending all the work every row in the table, but the user is unlikely to click on any particular row.